lib/data/datasource/remote/api_service.dart

```diff
@@ -0,0 +1,137 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+
+import '../../../core/util/mapper/mapper.dart';
+import '../../../domain/entity/genre.dart';
+import '../../../domain/entity/movie.dart';
+import '../../model/data_state.dart';
+import '../../model/genre_model.dart';
+
+import '../../model/now_playing_movies_model.dart';
+import '../../model/popular_movies_model.dart';
+import '../../model/top_rated_movie_model.dart';
+import '../../model/upcoming_movies_model.dart';
+
+class ApiService {
+  static const String baseUrl = 'https://api.themoviedb.org/3/movie/';
+  static const String authToken =
+      'Bearer eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiI5YTY1ZjQ0YjIzYWQ0MDU0NmRhODlhNjgxZDA3NGUzMSIsInN1YiI6IjY0ZmIyOTcxYWY1OGNiMGE2MmU4M2NlZCIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.laM3iXSCw-7DQbeVAx6nimMtHj3RfTq-XtmSiQoMcPI';
+  static const String genresUrl =
+      'https://api.themoviedb.org/3/genre/movie/list';
+  static const String topRated = 'top_rated';
+  static const String popular = 'popular';
+  static const String nowPlaying = 'now_playing';
+  static const String upcoming = 'upcoming';
+  static const String errorMsg = 'error while fetching data';
+
+  Future<DataState<List<Movie>>> getPopularMovies() async {
+    try {
+      final response = await http.get(
+        Uri.parse('$baseUrl$popular'),
+        headers: {HttpHeaders.authorizationHeader: authToken},
+      );
+      if (response.statusCode == 200) {
```

Comment:
Instead to use 200 you can use HttpStatus.OK

Reply:
done